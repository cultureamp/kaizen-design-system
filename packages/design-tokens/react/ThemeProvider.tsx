import React from "react"
import { ThemeManager, defaultTheme, Theme } from "../"

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager: ThemeManager
  children: React.ReactNode
}) => {
  const [theme, setTheme] = React.useState<Theme>(
    themeManager.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme) => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManager.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManager.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <ThemeContext.Provider value={theme}>
      {props.children}
    </ThemeContext.Provider>
  )
}

export const useTheme = () => React.useContext(ThemeContext)
