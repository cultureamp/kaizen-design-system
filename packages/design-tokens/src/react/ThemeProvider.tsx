import React, { useState } from "react"
import { ThemeManager, defaultTheme, Theme } from ".."

export const ThemeContext = React.createContext<Theme>(defaultTheme)

/**
 * Wrap your application in this provider using a ThemeManager, to synchronise it with a react context.
 * This allows child react elements to more easily use theme variables, using the {@link useTheme} hook.
 */
/**
 * @deprecated ThemeProvider is deprecated. Please use KaizenProvider from "@kaizen/components" instead.
 */
export const ThemeProvider = ({
  themeManager,
  ...props
}: {
  themeManager?: ThemeManager
  children: React.ReactNode
}): JSX.Element => {
  const [themeManagerInstance] = useState(
    () => themeManager || new ThemeManager(defaultTheme)
  )

  const [theme, setTheme] = React.useState<Theme>(
    themeManagerInstance.getCurrentTheme()
  )
  React.useEffect(() => {
    let cancelled = false
    const listener = (newTheme: Theme): void => {
      if (!cancelled) setTheme(newTheme)
    }
    themeManagerInstance.addThemeChangeListener(listener)
    return () => {
      cancelled = true
      themeManagerInstance.removeThemeChangeListener(listener)
    }
  }, [])

  return (
    <>
      <ThemeContext.Provider value={theme}>
        {props.children}
      </ThemeContext.Provider>
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-demi-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-medium.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/inter/inter-regular.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
      <link
        rel="preload"
        href="https://d1e7r7b0lb8p4d.cloudfront.net/fonts/tiempos/tiempos-headline-bold.woff2"
        as="font"
        type="font/woff2"
        crossOrigin="anonymous"
      />
    </>
  )
}

export const useTheme = (): Theme => React.useContext(ThemeContext)
