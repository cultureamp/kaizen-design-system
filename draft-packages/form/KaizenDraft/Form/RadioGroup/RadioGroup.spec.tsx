import React from "react"
import { render, screen } from "@testing-library/react"
import { RadioGroup } from "./index"

describe("<RadioGroup />", () => {
  describe("presentational", () => {
    it("renders a title", () => {
      const title = "Label"

      const { queryByText } = render(<RadioGroup labelText={title} />)
      expect(queryByText(title)).toBeTruthy()
    })
  })

  describe("accessibilty", () => {
    it("has an accessible name when not provided a labelId", () => {
      render(<RadioGroup labelText="Label title" />)
      expect(
        screen.getByRole("radiogroup", { name: "Label title" })
      ).toBeInTheDocument()
    })

    it("has an accessible name when provided a labelId", () => {
      render(<RadioGroup labelText="Label title" labelId="some-custom-id" />)
      expect(
        screen.getByRole("radiogroup", { name: "Label title" })
      ).toBeInTheDocument()
    })
  })
})
