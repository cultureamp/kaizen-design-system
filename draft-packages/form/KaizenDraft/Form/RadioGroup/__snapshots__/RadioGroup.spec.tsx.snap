// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup />  snapshots renders RadioGroup in a column and with a label 1`] = `
<div
  aria-labelledby="ba333b4a-403d-4576-8187-9b73fd89eca1"
  class="radioGroupContainer"
  data-automation-id=""
  role="radiogroup"
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id=""
      id="ba333b4a-403d-4576-8187-9b73fd89eca1"
    >
      <span
        class="appendedLabel"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup />  snapshots renders RadioGroup in a column and with a label 2`] = `
<div
  aria-labelledby="6aa19717-7bf2-41c7-bc79-5ae68a6be4b1"
  class="radioGroupContainer"
  data-automation-id=""
  role="radiogroup"
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id=""
      id="6aa19717-7bf2-41c7-bc79-5ae68a6be4b1"
    >
      <span
        class="appendedLabel"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup />  snapshots renders RadioGroup in a row and with a label 1`] = `
<div
  aria-labelledby="6ff1ed49-0d00-4c18-9953-8cfd0bbf0551"
  class="radioGroupContainer"
  data-automation-id=""
  role="radiogroup"
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id=""
      id="6ff1ed49-0d00-4c18-9953-8cfd0bbf0551"
    >
      <span
        class="appendedLabel"
      >
        Label
      </span>
    </label>
  </div>
</div>
`;

exports[`<RadioGroup />  snapshots renders RadioGroup with radios 1`] = `
<div
  aria-labelledby="b3b749ed-2aba-4c46-8119-f6235b1de8f3"
  class="radioGroupContainer"
  data-automation-id=""
  role="radiogroup"
>
  <div
    class="radioGroupLabel"
  >
    <label
      class="label text"
      data-automation-id=""
      id="b3b749ed-2aba-4c46-8119-f6235b1de8f3"
    >
      <span
        class="appendedLabel"
      >
        Label
      </span>
    </label>
  </div>
  <div
    class="container"
  >
    <label
      class="label radio"
      data-automation-id="radio-1-field-label"
      for="radio-1"
      id="radio-1-field-label"
    >
      <span>
        <input
          class="radioInput"
          data-automation-id="radio-1-radio-input"
          id="radio-1"
          name="radio"
          readonly=""
          type="radio"
          value="radio-1"
        />
        <span
          class="box"
        />
      </span>
      <span
        class="appendedLabel"
      >
        Label 1
      </span>
    </label>
  </div>
  <div
    class="container"
  >
    <label
      class="label radio"
      data-automation-id="radio-2-field-label"
      for="radio-2"
      id="radio-2-field-label"
    >
      <span>
        <input
          class="radioInput"
          data-automation-id="radio-2-radio-input"
          id="radio-2"
          name="radio"
          readonly=""
          type="radio"
          value="radio-2"
        />
        <span
          class="box"
        />
      </span>
      <span
        class="appendedLabel"
      >
        Label 2
      </span>
    </label>
  </div>
</div>
`;
